src/ddprof_worker.cc

```diff
@@ -88,9 +88,26 @@ DDRes report_lost_events(DDProfContext &ctx) {
   return {};
 }
 
+inline int64_t get_mono_to_realtime_offset(
+    const std::chrono::steady_clock::time_point &steady_now) {
```

Comment:
I think the logic is wrong here:
std::chrono::steady_clock is the same `CLOCK_MONOTONIC` so you are basically computing the offset between `CLOCK_MONOTONIC_RAW` and `CLOCK_MONOTONIC`.
You should use `std::chrono::system_clock`.
Also you should not use `CLOCK_MONOTONIC_RAW` but `PerfClock` instead: `PerfClock` is defined as the clock used for events from perf.
You could represent the offset as the epoch of PerfClock in the `system_clock` referential:
```
return std::chrono::system_clock::now() - PerfClock::now().time_since_epoch();
```

Reply:
Ah, good point.  Indeed, this seems fundamentally wrong.  Will review your notes.  Strong agree that `PerfClock` is better to use, now that I see it, since at the very least there's a stronger guarantee (thanks to your hard work!!!) that this is the correct underlying clock for the moment.

Reply:
OK, this was a good suggestion and I changed the strategy to use the PerfClock.  Confirmed that the offsets are better.  Thank you!

Reply:
Fixed in 1a39a1c